AsyncSwift/Observed/KeyChain.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  KeyChain.swift
+//  AsyncSwift
+//
+//  Created by Inho Choi on 2022/09/15.
+//
+
+import Foundation
+
+final class KeyChain {
+    static let shared = KeyChain()
+
+    private init() { }
+
+    func addItem(key: Any, pwd: Any) -> Bool {
+        let addQuery: [CFString: Any] = [kSecClass: kSecClassGenericPassword,
+                                         kSecAttrAccount: key,
+                                         kSecValueData: (pwd as AnyObject).data(using: String.Encoding.utf8.rawValue) as Any]
+
+        let result: Bool = {
+            let status = SecItemAdd(addQuery as CFDictionary, nil)
+            if status == errSecSuccess {
+                return true
+            } else if status == errSecDuplicateItem {
+                return updateItem(value: pwd, key: key)
+            }
+
+            print("addItem Error : \(status.description))")
+            return false
+        }()
+
+        return result
+    }
+
+    func getItem(key: Any) -> Any? {
+        let getQuery: [CFString: Any] = [kSecClass: kSecClassGenericPassword,
+                                         kSecAttrAccount: key,
+                                         kSecReturnAttributes: true,
+                                         kSecReturnData: true]
+        var item: CFTypeRef?
+        let result = SecItemCopyMatching(getQuery as CFDictionary, &item)
+
+        if result == errSecSuccess {
+            if let existingItem = item as? [String: Any],
+                let data = existingItem[kSecValueData as String] as? Data,
+                let password = String(data: data, encoding: .utf8) {
```

Comment:
```suggestion
        if result == errSecSuccess, 
        let existingItem = item as? [String: Any], 
        let data = existingItem[kSecValueData as String] as? Data, 
        let password = String(data: data, encoding: .utf8) {
```

하나의 조건문으로 return 구현이 가능해보이네요

Reply:
수정하겠습니다!